### ISSUE TITLE
About Loss Function

### ISSUE THREAD

**Issue by Tim-Hung** [TorchSSL/TorchSSL#24]
Hi, Thanks for your great work first!

I'm just wondering why you use `nll_loss ` instead of `cross_entropy` in your code [https://github.com/TorchSSL/TorchSSL/blob/c4d272202b254efd85e78c9f72c8842edc752e04/train_utils.py#L295](url)

BTW, I see `cross_entropy` was commented and said it is unstable, can you share your experience about why it is unstable?

For my understanding, CrossEntropyLoss = softmax + log + nll_Loss.

If I'm missing something, please forgive me. Thanks!

**Comment by qianlanwyd** [comment]
I think NLL loss is more numerical steady than cross entropy cause it calculate log softmax instead of softmax first. For more information, you can check https://stats.stackexchange.com/questions/436766/cross-entropy-with-log-softmax-activation

**Comment by Tim-Hung** [comment]
Thanks for your reply.
I'll check it in my experiment.
